Assets/Scripts/Region/Region.cs

```diff
@@ -1,24 +1,36 @@
 using UnityEngine;
 
-public class Region : MonoBehaviour {
+using Core;
+
+public class Region : MonoBehaviour, Actor.iGuid {
+    [SerializeField] bool debug;
     [SerializeField] SpriteRenderer sr;
     [SerializeField] Color inactiveColor;
     [SerializeField] Color activeColor;
 
-    void Start() {
-        // sr.enabled = false;
-    }
+    [Space]
+
+    [SerializeField] Core.EventChannelSO eventChannel;
 
-    void Update() {
+    // props
+    System.Guid guid = new System.Guid(); // this is the unique ID used for comparing enemies, bosses, pickups, destructibles etc.
 
+    public System.Guid Guid() {
+        return guid;
+    }
+
+    void Start() {
+        if (!debug) sr.enabled = false;
     }
 
     void Activate() {
         sr.color = activeColor;
+        eventChannel.OnRegionActivate.Invoke(guid);
```

Comment:
This will communicate with each enemy that belongs to a region in order to enable/disable that enemy when traversing from one region to another.

Reply:
When each enemy spawns, its `Awake` method will call and it will find out which region it is inside.

With this design decision, there is a very small chance of a race condition, where the enemy hasn't finished registering its region when the `OnRegionDeactivate` event fires; in this case we may opt to optimistically remove the enemy anyways. But this is likely over-engineering at this point.